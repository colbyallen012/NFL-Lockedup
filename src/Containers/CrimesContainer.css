.crime-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.heading {
  margin: auto;
  text-align: center;
  width: 300px;
  position: relative;
}

.all-container {
  display: flex;
  flex-direction: row;
}

.crimes-container, .player-container, .position-container {
  background-size: initial;
  color: white;
  display: flex;
  flex-direction: column;
  height: fit-content;
  overflow: scroll;
  height: 300px;
  width: 420px;
}

.heading {
  color: black;
}