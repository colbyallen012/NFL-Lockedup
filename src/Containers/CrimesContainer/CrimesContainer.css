.crime-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.heading {
  margin: auto;
  text-align: center;
  width: 300px;
  position: relative;
}

.all-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: 5px auto;
  width: 90%;
}

.crimes-container, .player-container, .position-container {
  background-size: initial;
  color: white;
  display: flex;
  flex-direction: column;
  height: fit-content;
  overflow: scroll;
  height: 300px;
  width: 420px;
}

.heading {
  color: black;
}