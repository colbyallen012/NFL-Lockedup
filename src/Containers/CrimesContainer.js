import React from 'react'; 
import Crime from '../Components/Crime'
import './CrimesContainer.css'

const CrimesContainer = ({crimes}) => {
  const displayCrimes = crimes.map(crime => {
    return (
      <Crime 
        category={crime.Category}
        arrests={crime.arrest_count}
        key={crime.category}
      />
    )
  })
  return (
    <div>
      <section className='crime-header'>
        <h2 className='heading'>All Crimes</h2>
        <h2 className='heading'>Crimes By Player</h2>  
        <h2 className='heading'>Crimes By Position</h2>      
      </section>
      <section className='crimes-container'>
        {displayCrimes}
      </section>
    </div>
  )
}

export default CrimesContainer;