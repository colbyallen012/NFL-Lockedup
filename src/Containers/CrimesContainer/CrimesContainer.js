import React from 'react'; 
import Crime from '../../Components/Crime/Crime'
import Player from '../../Components/Player/Player'
import Position from '../../Components/Position/Position'
import './CrimesContainer.css'

const CrimesContainer = ({crimes, players, positions}) => {
  const displayCrimes = crimes.map(crime => {
    return (
      <Crime 
        category={crime.Category}
        arrests={crime.arrest_count}
        key={crime.category}
      />
    )
  })

  const displayPlayers = players.map(player => {
    return (
      <Player 
        name={player.Name}
        city={player.Team}
        team={player.Team_name}
        position={player.Position}
        arrests={player.arrest_count}
      />
    )
  })

  const displayPositions = positions.map(position => {
    return (
      <Position
        position={position.Position}
        arrests={position.arrest_count}
      />
    )
  })

  return (
    <div>
      <section className='all-container'>
        <article className='crimes-container'>
          <h2 className='heading'>By Crime</h2>
          {displayCrimes}
        </article>
        <article className='player-container'>
          <h2 className='heading'>Multiple Offenders</h2>  
          {displayPlayers}
        </article>
        <article className='position-container'>
          <h2 className='heading'>By Position</h2>  
          {displayPositions}
        </article>
      </section>
    </div>
  )
}

export default CrimesContainer;