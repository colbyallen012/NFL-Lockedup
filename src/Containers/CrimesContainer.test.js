import { shallow } from 'enzyme';
import React from 'react';
import CrimesContainer from './CrimesContainer';

describe('CrimesContainer', () => {
  let wrapper;
  const mockCrimes = [{
    Category: "DUI",
    arrest_count: "224"
    },
    {
    Category: "Drugs",
    arrest_count: "107"
  }]

  const mockPlayers = [{
    Name: "Kenny Britt",
    Team: "TEN",
    Team_name: "Titans",
    Team_city: "Nashville",
    Position: "WR",
    arrest_count: "7"
  }]

  const mockPositions = [{
    Position: "WR",
    arrest_count: "148"
    },
    {
    Position: "LB",
    arrest_count: "127"
  }]

  beforeEach(() => {
    wrapper = shallow(<CrimesContainer crimes={mockCrimes} players={mockPlayers} positions={mockPositions} key={1}/>)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  }); 
})