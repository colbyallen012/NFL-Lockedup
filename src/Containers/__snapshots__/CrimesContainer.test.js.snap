// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrimesContainer should match snapshot 1`] = `
<div>
  <section
    className="all-container"
  >
    <article
      className="crimes-container"
    >
      <h2
        className="heading"
      >
        By Crime
      </h2>
      <Crime
        arrests="224"
        category="DUI"
      />
      <Crime
        arrests="107"
        category="Drugs"
      />
    </article>
    <article
      className="player-container"
    >
      <h2
        className="heading"
      >
        Multiple Offenders
      </h2>
      <Player
        arrests="7"
        city="TEN"
        name="Kenny Britt"
        position="WR"
        team="Titans"
      />
    </article>
    <article
      className="position-container"
    >
      <h2
        className="heading"
      >
        By Position
      </h2>
      <Player
        arrests="148"
        position="WR"
      />
      <Player
        arrests="127"
        position="LB"
      />
    </article>
  </section>
</div>
`;
